rs/tests/networking/cloner_canisters_workload.rs

```diff
@@ -0,0 +1,224 @@
+//!
+//! Use the console link to stop and start the replica process in order to start a catch up process.
+//! To stop the replica use the command:
+//! `systemctl stop ic-replica`
+//!
+//! To start the replica again, use the command:
+//! `systemctl start ic-replica`
+
+use anyhow::Result;
+use candid::{CandidType, Encode};
+use ic_registry_subnet_type::SubnetType;
+use ic_system_test_driver::{
+    canister_agent::{CanisterAgent, HasCanisterAgentCapability},
+    driver::{
+        farm::HostFeature,
+        group::SystemTestGroup,
+        ic::{AmountOfMemoryKiB, ImageSizeGiB, InternetComputer, NrOfVCPUs, Subnet, VmResources},
+        prometheus_vm::{HasPrometheus, PrometheusVm},
+        test_env::TestEnv,
+        test_env_api::{HasPublicApiUrl, HasTopologySnapshot, HasWasm, IcNodeContainer},
+    },
+    systest,
+};
+use serde::{Deserialize, Serialize};
+use slog::info;
+use std::time::Duration;
+
+// 2 hours
+const WORKLOAD_RUNTIME: Duration = Duration::from_secs(2 * 60 * 60);
+
+// 10 minutes
+const DOWNLOAD_PROMETHEUS_WAIT_TIME: Duration = Duration::from_secs(10 * 60);
+
+// Timeout parameters
+const TASK_TIMEOUT_DELTA: Duration = Duration::from_secs(3600);
+const OVERALL_TIMEOUT_DELTA: Duration = Duration::from_secs(3600);
+
+const CLONER_CANISTER_WASM: &str = "rs/tests/src/cloner_canister.wasm.gz";
+const COUNTER_CANISTER_WAT: &str = "rs/tests/src/counter.wat";
+
+const SUBNET_SIZE: usize = 13;
+const INITIAL_NOTARY_DELAY: Duration = Duration::from_millis(200);
+
+// 100,000 canisters, with 500 batches, will take ~25 minutes to set up.
+// Yields 280-310ms commit and certify times.
+// We need minimum 350+ms, so we should probably push this to 150,000 canisters.
+const NUMBER_OF_CANISTERS_TO_INSTALL: u64 = 125_000;
+const CANISTERS_INSTALLED_PER_CLONER_CANISTER: u64 = 500;
+const AMOUNT_OF_CLONER_CANISTERS: u64 =
+    NUMBER_OF_CANISTERS_TO_INSTALL / CANISTERS_INSTALLED_PER_CLONER_CANISTER;
+
+// Don't change these values.
+/// The batch size for the cloner canister to install canisters concurrently.
+const CLONER_CANISTER_BATCH_SIZE: u64 = 250;
+/// The initial cycles for the canisters spun up by the cloner canister.
+/// If too low or too high, the canisters will not be installed, or the cloner will run out of cycles.
+const INITIAL_CYCLES: u64 = 10_u64.pow(11); // 100B Cycles
+
+fn main() -> Result<()> {
+    let per_task_timeout: Duration = WORKLOAD_RUNTIME + TASK_TIMEOUT_DELTA;
+    let overall_timeout: Duration = per_task_timeout + OVERALL_TIMEOUT_DELTA;
+    SystemTestGroup::new()
+        .with_setup(config)
+        .add_test(systest!(install_cloner_canisters))
+        .with_timeout_per_test(per_task_timeout) // each task (including the setup function) may take up to `per_task_timeout`.
+        .execute_from_args()?;
+    Ok(())
+}
+
+pub fn config(env: TestEnv) {
```

Comment:
The modern naming for setup function is `setup` instead of `config`:
```suggestion
        .with_setup(setup)
        .add_test(systest!(install_cloner_canisters))
        .with_timeout_per_test(per_task_timeout) // each task (including the setup function) may take up to `per_task_timeout`.
        .execute_from_args()?;
    Ok(())
}

pub fn setup(env: TestEnv) {
```

Reply:
TIL